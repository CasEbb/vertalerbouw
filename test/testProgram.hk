skrieben('i','n','t');
getalletjen ivar;
ivar := {
    getalletjen ivar1, ivar2;
    vraogen(ivar1, ivar2);
    skrieben(ivar1, ivar2);
    
    blievend getalletjen iconst1 = 1;
    blievend getalletjen iconst2 = 2;
    
    ivar2 := ivar1 := +18 + 2 * -9;
    skrieben(iconst1 * iconst2 > ivar2 - ivar1,
        ivar1 < ivar2 && iconst1 <= iconst2);
    ivar1 < vraogen(ivar2) && iconst1 <= iconst2;
    ivar2 := skrieben(ivar2) + 10;
} + 5;

skrieben('b','o','o','l');
walofneet bvar;
bvar := {
    walofneet bvar;
    vraogen(bvar);
    skrieben(bvar);
    bvar := 12 / 5 * 5 + 12 % 5 == 12 && 6 >= 6;
    blievend walofneet bconst = wal;
    skrieben(!neet && bvar == bconst || wal <> neet);
} && wal;

skrieben('c','h','a','r');
lettertjen cvar;
cvar := {
    lettertjen cvar1, cvar2;
    vraogen(cvar1);
    blievend lettertjen cconst = 'c';
    cvar2 := 'z';
    skrieben('a', cvar1 == cconst && (cvar2 <> 'b' || !wal));
    'b';
};

skrieben(ivar, bvar, cvar);
