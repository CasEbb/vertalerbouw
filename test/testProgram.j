.class public testProgram
.super java/lang/Object

.method public static main([Ljava/lang/String;)V
    .limit stack 50
    .limit locals 100
    new java/util/Scanner
    dup
    getstatic java/lang/System/in Ljava/io/InputStream;
    invokespecial java/util/Scanner/<init>(Ljava/io/InputStream;)V
    astore_1
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 105
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 110
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 116
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    invokevirtual java/io/PrintStream/println()V
    iconst_0
    istore 2
    iconst_0
    istore 3
    iconst_0
    istore 4
    aload_1
    invokevirtual java/util/Scanner/nextInt()I
    istore 3
    aload_1
    invokevirtual java/util/Scanner/nextInt()I
    istore 4
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iload 3
    invokevirtual java/io/PrintStream/print(I)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iload 4
    invokevirtual java/io/PrintStream/print(I)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    invokevirtual java/io/PrintStream/println()V
    bipush 1
    istore 5
    bipush 2
    istore 6
    bipush 18
    bipush 2
    bipush 9
    ineg
    imul
    iadd
    istore 3
    iload 3
    istore 4
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iload 5
    iload 6
    imul
    iload 4
    iload 3
    isub
    if_icmpgt L0
    iconst_0
    goto L0r
    L0:
    iconst_1
    L0r:
    invokevirtual java/io/PrintStream/print(I)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iload 3
    iload 4
    if_icmplt L1
    iconst_0
    goto L1r
    L1:
    iconst_1
    L1r:
    iload 5
    iload 6
    if_icmple L2
    iconst_0
    goto L2r
    L2:
    iconst_1
    L2r:
    iand
    invokevirtual java/io/PrintStream/print(I)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    invokevirtual java/io/PrintStream/println()V
    iload 3
    aload_1
    invokevirtual java/util/Scanner/nextInt()I
    istore 4
    iload 4
    if_icmplt L3
    iconst_0
    goto L3r
    L3:
    iconst_1
    L3r:
    iload 5
    iload 6
    if_icmple L4
    iconst_0
    goto L4r
    L4:
    iconst_1
    L4r:
    iand
    pop
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iload 4
    invokevirtual java/io/PrintStream/print(I)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    invokevirtual java/io/PrintStream/println()V
    iload 4
    bipush 10
    iadd
    istore 4
    iload 4
    bipush 5
    iadd
    istore 2
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 98
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 111
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 111
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 108
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    invokevirtual java/io/PrintStream/println()V
    iconst_0
    istore 3
    iconst_0
    istore 4
    aload_1
    invokevirtual java/util/Scanner/nextInt()I
    istore 4
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iload 4
    invokevirtual java/io/PrintStream/print(I)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    invokevirtual java/io/PrintStream/println()V
    bipush 12
    bipush 5
    idiv
    bipush 5
    imul
    bipush 12
    bipush 5
    irem
    iadd
    bipush 12
    if_icmpeq L5
    iconst_0
    goto L5r
    L5:
    iconst_1
    L5r:
    bipush 6
    bipush 6
    if_icmpge L6
    iconst_0
    goto L6r
    L6:
    iconst_1
    L6r:
    iand
    istore 4
    bipush 1
    istore 5
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 0
    ifeq L7
    iconst_0
    goto L7r
    L7:
    iconst_1
    L7r:
    iload 4
    iload 5
    if_icmpeq L8
    iconst_0
    goto L8r
    L8:
    iconst_1
    L8r:
    iand
    bipush 1
    bipush 0
    if_icmpne L9
    iconst_0
    goto L9r
    L9:
    iconst_1
    L9r:
    ior
    invokevirtual java/io/PrintStream/print(I)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    invokevirtual java/io/PrintStream/println()V
    bipush 0
    ifeq L7
    iconst_0
    goto L7r
    L7:
    iconst_1
    L7r:
    iload 4
    iload 5
    if_icmpeq L8
    iconst_0
    goto L8r
    L8:
    iconst_1
    L8r:
    iand
    bipush 1
    bipush 0
    if_icmpne L9
    iconst_0
    goto L9r
    L9:
    iconst_1
    L9r:
    ior
    bipush 1
    iand
    istore 3
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 99
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 104
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 97
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 114
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    invokevirtual java/io/PrintStream/println()V
    iconst_0
    istore 4
    iconst_0
    istore 5
    iconst_0
    istore 6
    aload_1
    invokevirtual java/util/Scanner/next()Ljava/lang/String;
    iconst_0
    invokevirtual java/lang/String/charAt(I)C
    istore 5
    bipush 99
    istore 7
    bipush 122
    istore 6
    getstatic java/lang/System/out Ljava/io/PrintStream;
    bipush 97
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iload 5
    iload 7
    if_icmpeq L10
    iconst_0
    goto L10r
    L10:
    iconst_1
    L10r:
    iload 6
    bipush 98
    if_icmpne L11
    iconst_0
    goto L11r
    L11:
    iconst_1
    L11r:
    bipush 1
    ifeq L12
    iconst_0
    goto L12r
    L12:
    iconst_1
    L12r:
    ior
    iand
    invokevirtual java/io/PrintStream/print(I)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    invokevirtual java/io/PrintStream/println()V
    bipush 98
    istore 4
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iload 2
    invokevirtual java/io/PrintStream/print(I)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iload 3
    invokevirtual java/io/PrintStream/print(I)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iload 4
    invokevirtual java/io/PrintStream/print(C)V
    getstatic java/lang/System/out Ljava/io/PrintStream;
    invokevirtual java/io/PrintStream/println()V
    aload_1
    invokevirtual java/util/Scanner/close()V
    return
.end method
Generated: testProgram.class
