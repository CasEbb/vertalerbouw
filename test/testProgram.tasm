LOADL       105
CALL        put
LOADL       110
CALL        put
LOADL       116
CALL        put
CALL        puteol
PUSH        1
PUSH        2
LOADA       1[SB]
CALL        getint
LOADA       2[SB]
CALL        getint
LOAD(1)     1[SB]
CALL        putint
LOAD(1)     2[SB]
CALL        putint
CALL        puteol
PUSH        1
LOADL       1
STORE(1)    3[SB]
PUSH        1
LOADL       2
STORE(1)    4[SB]
LOADL       18
LOADL       2
LOADL       9
CALL        neg
CALL        mult
CALL        add
STORE(1)    1[SB]
LOAD(1)     1[SB]
STORE(1)    2[SB]
LOAD(1)     3[SB]
LOAD(1)     4[SB]
CALL        mult
LOAD(1)     2[SB]
LOAD(1)     1[SB]
CALL        sub
CALL        gt
CALL        putint
LOAD(1)     1[SB]
LOAD(1)     2[SB]
CALL        lt
LOAD(1)     3[SB]
LOAD(1)     4[SB]
CALL        le
CALL        and
CALL        putint
CALL        puteol
LOAD(1)     1[SB]
LOADA       2[SB]
CALL        getint
LOAD(1)     2[SB]
CALL        lt
LOAD(1)     3[SB]
LOAD(1)     4[SB]
CALL        le
CALL        and
POP(0)      1
LOAD(1)     2[SB]
CALL        putint
CALL        puteol
LOAD(1)     2[SB]
LOADL       10
CALL        add
STORE(1)    2[SB]
LOAD(1)     2[SB]
POP(1)      4
LOADL       5
CALL        add
STORE(1)    0[SB]
LOADL       98
CALL        put
LOADL       111
CALL        put
LOADL       111
CALL        put
LOADL       108
CALL        put
CALL        puteol
PUSH        1
PUSH        1
LOADA       2[SB]
CALL        getint
LOAD(1)     2[SB]
CALL        putint
CALL        puteol
LOADL       12
LOADL       5
CALL        div
LOADL       5
CALL        mult
LOADL       12
LOADL       5
CALL        mod
CALL        add
LOADL       12
LOADL       1
CALL        eq
LOADL       6
LOADL       6
CALL        ge
CALL        and
STORE(1)    2[SB]
PUSH        1
LOADL       1
STORE(1)    3[SB]
LOADL       0
CALL        not
LOAD(1)     2[SB]
LOAD(1)     3[SB]
LOADL       1
CALL        eq
CALL        and
LOADL       1
LOADL       0
LOADL       1
CALL        ne
CALL        or
CALL        putint
CALL        puteol
LOADL       0
CALL        not
LOAD(1)     2[SB]
LOAD(1)     3[SB]
LOADL       1
CALL        eq
CALL        and
LOADL       1
LOADL       0
LOADL       1
CALL        ne
CALL        or
POP(1)      2
LOADL       1
CALL        and
STORE(1)    1[SB]
LOADL       99
CALL        put
LOADL       104
CALL        put
LOADL       97
CALL        put
LOADL       114
CALL        put
CALL        puteol
PUSH        1
PUSH        2
LOADA       3[SB]
CALL        get
PUSH        1
LOADL       99
STORE(1)    5[SB]
LOADL       122
STORE(1)    4[SB]
LOADL       97
CALL        put
LOAD(1)     3[SB]
LOAD(1)     5[SB]
LOADL       1
CALL        eq
LOAD(1)     4[SB]
LOADL       98
LOADL       1
CALL        ne
LOADL       1
CALL        not
CALL        or
CALL        and
CALL        putint
CALL        puteol
LOADL       98
POP(1)      3
STORE(1)    2[SB]
LOAD(1)     0[SB]
CALL        putint
LOAD(1)     1[SB]
CALL        putint
LOAD(1)     2[SB]
CALL        put
CALL        puteol
HALT
