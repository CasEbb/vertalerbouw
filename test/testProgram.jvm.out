int
12
10
3
bool
1
char
a1
181b
